### DISCUSSION TITLE
Airflow Docker Vulnerability Scan

### DISCUSSION THREAD

**Discussion by jadsultan123** [apache/airflow#28408] [General]
Hello,

we made a Security scan for apache/airflow:2.5.0, and we found 207 vulnerabilities on scan on Harbor Software.

Is there any solution to remove these vulnerabilities by rebuilding or extending the docker image?

Thank you

**Comment by potiuk** [comment]
Did you read https://airflow.apache.org/docs/docker-stack/build.html (this doc is referred to in the image as documentation URL in case you missed it). 

But there is no guarantee all issues are removed this way - it all depends on many factors - base image and released software. Our image is a reference one and you can freelly extend and customize as you wish. If you want to fix things on your own - you can always modify the Dockerfile (only Dockerfile is needed to rebuild the image). Specifically https://airflow.apache.org/docs/docker-stack/build.html#customizing-the-image chapter and examples explain how to build your image from the scratch.

BTW. Relying on scanners is something that you have to deal with if you want to clear all reported software. The fact that scanner reports things, does not matter that the  vulnerability is exploitable - the number of false positives is high. If you chose that way, you need to deal with those false positives. 

But if your company (evidently security-minded) would like to help and analyse those vulnerabilities and report any exploitable one (and report a reproducible vulnerability report  - this is one of the best ways you can give back to the community for the free software you use.  The security policy and ways to report such issues is here: https://github.com/apache/airflow/security/policy

We keep up with latest compatible versions released of both when we release new airflow version, so the easiest things to do to keep up is to upgrade to latest version of Airlfow immediately after it is releaased. If you choose to stay with older version - we do not update them and you need to do it on your own.
